--- sources/DTN2/servlib/bundling/BundleProtocol.cc	2013-05-24 20:22:12.000000000 +0200
+++ dtn-2.9.0/servlib/bundling/BundleProtocol.cc	2012-12-23 22:42:28.000000000 +0100
@@ -110,15 +110,6 @@
 //----------------------------------------------------------------------
 void
 BundleProtocol::reload_post_process(Bundle* bundle)
@@ -213,12 +204,13 @@
          ++iter)
     {
         log_debug_p(LOG, "adding api_block");
-        if(BP_FAIL == iter->owner()->prepare(bundle, xmit_blocks, 
+        /*if(BP_FAIL == iter->owner()->prepare(bundle, xmit_blocks, 
                                &*iter, link, BlockInfo::LIST_API)){
 
             log_err_p(LOG, "BundleProtocol::prepare_blocks: %d->prepare returned BP_FAIL on bundle %d", iter->owner()->block_type(), bundle->bundleid());
             goto fail;
-        }
+        }*/
+        iter->owner()->prepare(bundle, xmit_blocks, &*iter, link, BlockInfo::LIST_API);
     }
 
     //+[AEB]
